engine/build.gradle

```diff
@@ -179,11 +181,13 @@ jar {
 
     from(tasks.getByName("cacheReflections").outputs)
 
-    manifest {
-        def manifestClasspath = "$subDirLibs/" + configurations."${sourceSets.main.runtimeClasspathConfigurationName}".collect {
-            it.getName()
-        }.join(" $subDirLibs/")
-        attributes("Class-Path": manifestClasspath, "Implementation-Title": "Terasology-" + project.name, "Implementation-Version": env.BUILD_NUMBER + ", " + convertGitBranch(env.GIT_BRANCH) + ", " + env.BUILD_ID + ", " + displayVersion)
+    doFirst {
```

Comment:
trying to collect RuntimeClasspath at setup gradle script is bad practics ( it happens every time when you press gradle's refresh button)
Also i catched race condition over gradle's cache  when extracted TypeHandlerLibrary.

Then I did use `doFirst` for avoid race condition and made collecting RuntimeClasspath task-related.
`doFirst`, not `doLast` - because manifest required at task's running.

 
{`@see` https://docs.gradle.org/current/userguide/tutorial_using_tasks.html}